<?xml version="1.0" encoding="UTF-8"?>

<robot name="sentryup">

    <link name="base_link">
        <visual>
            <geometry>
                <box size="0.5 1 0.16" />
            </geometry>
        </visual>
    </link>

    <link name="gimbal_yaw">
        <visual>
            <geometry>
                <box size="0.1 0.1 0.5" />
            </geometry>
            <origin xyz="0 0 0.081" rpy="0 0 0" />
        </visual>
    </link>

    <joint name="yaw_joint" type="continuous">
        <parent link="base_link" />
        <child link="gimbal_yaw" />
        <origin xyz="0 0 0.081" rpy="0 0 0" />
        <axis xyz="0 0 1" />
    </joint>

    <link name="gimbal_pitch">
        <visual>
            <geometry>
                <box size="0.5 0.1 0.1" />
            </geometry>
            <origin rpy="0 0 0" xyz="0.149 0 0" />
        </visual>
    </link>

    <joint name="pitch_joint" type="continuous">
        <parent link="gimbal_yaw" />
        <child link="gimbal_pitch" />
        <origin xyz="0 0 0.149" rpy="0 0 0" />
        <axis xyz="0 1 0" />
    </joint>

    <link name="gimbal">

    </link>

    <joint name="gimbal_joint" type="fixed">
        <parent link="gimbal_pitch" />
        <child link="gimbal" />
        <origin xyz="0.168 0 0" rpy="-1.57 0 -1.57" />
    </joint>

</robot>